client/extensions/woocommerce/state/ui/shipping/zones/methods/reducer.js

```diff
@@ -38,15 +43,100 @@ reducer[ WOOCOMMERCE_SHIPPING_ZONE_METHOD_ADD ] = ( state, action ) => {
 	const id = nextBucketIndex( state.creates );
 	let method = { id, methodType, enabled: true };
 	if ( builtInShippingMethods[ methodType ] ) {
-		method = { ...method, ...builtInShippingMethods[ methodType ]( undefined, action ) };
+		method = {
+			...method,
+			...builtInShippingMethods[ methodType ]( undefined, action ),
+			title: getMethodName( methodType ),
+		};
 	}
 	return { ...state,
 		creates: [ ...state.creates, method ],
+		currentlyEditingId: id,
+		currentlyEditingChanges: { isNew: true },
```

Comment:
I think that `isNew` should be at the same level as `currentlyEditingId` and `currentlyEditingChanges`. Right now you're merging `currentlyEditingChanges` with the "comitted" changes on close, and that would copy the `isNew` flag too (which is useless).